src/shims/env.rs

```diff
@@ -0,0 +1,60 @@
+use std::collections::HashMap;
+
+use rustc::ty::layout::{Size, Align};
+use rustc_mir::interpret::{Pointer, Memory};
+use crate::stacked_borrows::Tag;
+use crate::*;
+
+#[derive(Default)]
+pub struct EnvVars {
+    map: HashMap<Vec<u8>, Pointer<Tag>>,
+}
+
+impl EnvVars {
+    pub(crate) fn init<'mir, 'tcx>(
+        ecx: &mut InterpCx<'mir, 'tcx, Evaluator<'tcx>>,
+        communicate: bool,
```

Comment:
Ah, I entirely forgot that the machine has this flag. So can you remove this parameter, and use `ecx.machine.communicate` instead?

Reply:
Since you are probably already asleep, I'll do this as part of https://github.com/rust-lang/miri/pull/909.